File: Task_1_1_2/src/main/java/ru/nsu/smolin/Renderer.java
package ru.nsu.smolin;

import java.util.List;

public final class Renderer {
    private Renderer() {}

    public static void showSplash() {

        System.out.println(
                "+----------------------------------------------------+\n" +
                        "|                                                    |\n" +
                        "|                   .---------.                      |\n" +
                        "|                   |A        |                      |\n" +
                        "|                   |         |                      |\n" +
                        "|                   |    " + Config.BLACK + "S" + Config.RESET + "    |                      |\n" +
                        "|                   |         |                      |\n" +
                        "|                   |        A|                      |\n" +
                        "|                   '---------'                      |\n" +
                        "|                                                    |\n" +
                        "|             " + Config.RED + "H" + " " + Config.BLACK + "S" + "   BLACKJACK  " + Config.BLACK + "T" + Config.RED + " " + "D" + Config.RESET +
                        "                   |\n" +
                        "|            Press" + " " + Config.RED + " ENTER" + Config.RESET + " to start...                |\n" +
                        "+----------------------------------------------------+"
        );
    }


    public static void showHands(List<Card> player, List<Card> dealer, boolean hideDealerHole) {
        String playerStr = joinShort(player);
        String dealerStr = hideDealerHole && dealer.size() >= 2
                ? dealer.get(0).shortString() + " [XX]"
                : joinShort(dealer);
        String dealerSum = hideDealerHole ? "" : " (sum: " + Scoring.total(dealer) + ")";
        System.out.println(
                "Your cards:   " + playerStr + "  (sum: " + Scoring.total(player) + ")\n" +
                        "Dealer cards: " + dealerStr + dealerSum + "\n" +
                        "Chose action: Take / Pass / Info   (or Exit)"
        );
    }

    public static void showRoundHeader(int n) {
        System.out.println("\n-------------- ROUND " + n + " --------------\n");
    }

    private static String joinShort(List<Card> hand) {
        StringBuilder sb = new StringBuilder();
        for (int i = 0; i < hand.size(); i++) {
            if (i > 0) sb.append(' ');
            sb.append(hand.get(i).shortString());
        }
        return sb.toString();
    }
}

File: Task_1_2_1_Kotlin/src/main/kotlin/ru/nsu/smolin/Renderer.kt
package ru.nsu.smolin

object Renderer {
    private const val RESET = "\u001B[0m"
    private const val RED = "\u001B[31m"
    private const val BLACK = "\u001B[30m"

    private fun colorWrap(text: String, color: String): String {
        return if(Config.useColors) "$color$text$RESET" else text
    }

    /**
     * Showing start display
     *
     */
    fun showSplash() {
        val redDiamond = colorWrap("♦", RED)
        val blackClub = colorWrap("♣", BLACK)
        val blackSpade = colorWrap("♠", BLACK)
        val redHeart = colorWrap("♥", RED)
        val banner = """
╔══════════════════════════════════════════════════════╗
║                                                      ║
║                      ┌─────────┐                     ║
║                      │A        │                     ║
║                      │         │                     ║
║                      │    $blackSpade    │                     ║
║                      │         │                     ║
║                      │        A│                     ║
║                      └─────────┘                     ║
║                                                      ║
║                $redDiamond $blackClub   BLACKJACK  $blackSpade $redHeart                  ║
║                                                      ║
║               Press${colorWrap(" ENTER", RED)} to start...                ║
╚══════════════════════════════════════════════════════╝
""".trimEnd()
        println(banner)
    }

    /**
     * Showing hands
     *
     */
    fun showHands(player: List<Card>, dealer: List<Card>, hideDealerHole: Boolean) {
        val playerStr = player.joinToString(" ") { it.short() }
        val dealerStr = if (hideDealerHole && dealer.size >= 2)
            dealer.first().short() + " [XX]" else dealer.joinToString(" ") { it.short() }
        val dealerSum = if (hideDealerHole) "" else " (сумма: ${Scoring.total(dealer)})"
        println(
            """
        |Ваши карты: $playerStr (сумма: ${Scoring.total(player)})
        |Карты дилера: $dealerStr$dealerSum
        |Выберите действие: Take / Pass / Info (или Exit)
        """.trimMargin()
        )
    }

    fun showRoundHeader(n: Int) {
        println("\n-------------- ROUND $n --------------\n")
    }
}